zrcs/zrc-3.md

```diff
@@ -0,0 +1,432 @@
+| ZRC | Title                        | Status   | Type     | Author                                                                               | Created (yyyy-mm-dd) | Updated (yyyy-mm-dd) |
+| --- | ---------------------------- | -------- | -------- | ------------------------------------------------------------------------------------ | -------------------- | -------------------- |
+| 2   | Standard for Fungible Tokens | Draft | Standard | Cameron Sajedi <cameron@starlingfoundries.com> | 2019-10-15           | 2021-02-08           |
+
+## I. What are Metatransactions?
+
+The concept of metatransactions is intended to allow wallets to sign incomplete transactions that represent the authorization to spend a certain amount of tokens. These are valuable to users who do not wish to go through KYC, etc to get a trivial amount of ZIL in order to pay for a transaction. Developers may choose to include this pattern to give an easier onboarding UX, or also to enable transactions to be paid for in the native token. This standard includes an example of ZRC-2 integrated with metatransactions, but there are many situations where authorizing an effect via a signed message is valuable.  The drawback is a metatransaction cannot easily be cancelled, so signers of metatransactions should view them as transactions that have already been processed, and if the relayer censors them they should re-send the same exact metatransaction to another relayer to void the original censored transaction.
+
+## II. Abstract
+
+ZRC-3 defines a basic supplement to another contract with the ability to perform transitions with metatransactions. In this provided standard we build on ZRC-2 to enable metatransaction transfers. 
+
+## III. Motivation
+Metatransactions provide an alternative flow for users just getting started with their cryptocurrency and wallets. It allows for gasless transactions, cutting out the barrier of getting ZIL from an exchange or via mining before a Dapp user can interact with the smart contract. It can also be seen as a way to defer the processing of transactions or even to guard a transition that requires multiparty authorization, although those applications are left to implementations and future standards. This contract adds a `metatransfer` transition to the existing ZRC-2. It does this in a way that maintains compatibility with the Operator and Mintable variants of ZRC-2 as well as the Zilswap exchange, and hopefully future OpFi tools.
```

Comment:
I will remove the optional designation, I suppose technically in order to comply with ZRC3 at least the `ChequeSend` implementation needs to be included, `ChequeVoid` will be left optional since in some cases the concerns of the vendor might take precedence over spender sovereignty. I wrote a blog post about ZRC3 which I will add a link to as well.